File: 프로그래머스/unrated/181894. 2의 영역/2의 영역.py
def solution(arr):
    minIndex , maxIndex = 100000, -100000
    answer = []
    for i in range(len(arr)):
        if arr[i] == 2:
            minIndex = min(minIndex, i)
            maxIndex = max(maxIndex, i)
        
    while minIndex <= maxIndex:
        answer.append(arr[minIndex])
        minIndex +=1
    
    if not answer:
        answer.append(-1)
    return answer

File: 프로그래머스/unrated/181894. 2의 영역/2의 영역.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> arr) {
    int minIndex = INT32_MAX, maxIndex = INT32_MIN;
    vector<int> answer;
    for(int i=0;i<arr.size();i++){
        if( arr[i]== 2){
            minIndex = min(i, minIndex);
            maxIndex = max(maxIndex, i);
        }
    }
    
    // cout << minIndex << " " << maxIndex << "\n";
    while(minIndex <= maxIndex){
        answer.push_back(arr[minIndex]);
        minIndex++;
    }
    if(answer.size() == 0){
        answer.push_back(-1);
    }
    return answer;
}